src/coreclr/vm/jitinterface.cpp

```diff
@@ -13234,207 +13253,245 @@ PCODE UnsafeJitFunctionWorker(EECodeGenManager *pJitMgr, CEECodeGenInfo *pJitInf
-        EEPOLICY_HANDLE_FATAL_ERROR_WITH_MESSAGE(COR_E_EXECUTIONENGINE, W("Failed to load JIT compiler"));
-#endif // ALLOW_SXS_JIT
-    }
+#endif // defined(TARGET_AMD64) || defined(TARGET_ARM64)
 
-#ifdef _DEBUG
-    // This is here so we can see the name and class easily in the debugger
+        // No overflow detected
+        return false;
+    }
+};
 
-    LPCUTF8 cls  = ftn->GetMethodTable()->GetDebugClassName();
-    LPCUTF8 name = ftn->GetName();
+#if defined(TARGET_AMD64)
+BOOL JumpStubOverflowCheck::g_fAllowRel32 = TRUE;
+#endif // TARGET_AMD64
 
+static void LogJitMethodBegin(MethodDesc* ftn)
+{
+    STANDARD_VM_CONTRACT;
+    _ASSERTE(ftn != NULL);
+#ifdef LOGGING
     if (ftn->IsNoMetadata())
     {
         if (ftn->IsILStub())
         {
-            LOG((LF_JIT, LL_INFO10000, "{ Jitting IL Stub }\n"));
+            LOG((LF_JIT, LL_INFO10000, "{ Jitting IL Stub (%p)\n", ftn));
         }
         else
         {
-            LOG((LF_JIT, LL_INFO10000, "{ Jitting dynamic method }\n"));
+            LOG((LF_JIT, LL_INFO10000, "{ Jitting dynamic method (%p)\n", ftn));
         }
     }
     else
     {
-        SString methodString;
-        if (LoggingOn(LF_JIT, LL_INFO10000))
-            TypeString::AppendMethodDebug(methodString, ftn);
-
-        LOG((LF_JIT, LL_INFO10000, "{ Jitting method (%p) %s %s\n", ftn, methodString.GetUTF8(), ftn->m_pszDebugMethodSignature));
+        LPCUTF8 cls  = ftn->GetMethodTable()->GetDebugClassName();
+        LPCUTF8 name = ftn->GetName();
+        LOG((LF_JIT, LL_INFO10000, "{ Jitting method (%p) %s::%s  %s\n", ftn, cls, name, ftn->m_pszDebugMethodSignature));
     }
-#endif // _DEBUG
+#endif // LOGGING
+}
+
+static void LogJitMethodEnd(MethodDesc* ftn)
+{
+    STANDARD_VM_CONTRACT;
+    _ASSERTE(ftn != NULL);
+#ifdef LOGGING
+    LOG((LF_JIT, LL_INFO10000, "Done Jitting method (%p) }\n", ftn));
+#endif // LOGGING
+}
 
-    MethodInfoHelperContext cxt{ ftn, ILHeader };
-    CORINFO_METHOD_INFO methodInfo;
-    getMethodInfoHelper(cxt, &methodInfo);
+// ********************************************************************
+//                  README!!
+// ********************************************************************
 
-    if (ILHeader == nullptr)
-        ILHeader = cxt.Header;
+// The reason that this is named UnsafeJitFunction is that this helper
+// method is not thread safe!  When multiple threads get in here for
+// the same pMD, ALL of them MUST return the SAME value.
+// To ensure that this happens you must call MakeJitWorker.
+// It creates a DeadlockAware list of methods being jitted and prevents us
+// from trying to jit the same method more that once.
+//
+// Calls to this method that occur to check if inlining can occur on x86,
+// are OK since they discard the return value of this method.
+PCODE UnsafeJitFunction(PrepareCodeConfig* config,
+                        _In_opt_ COR_ILMETHOD_DECODER* ILHeader,
+                        _Out_ CORJIT_FLAGS* pJitFlags,
```

Comment:
```suggestion
                        _Out_ bool* pfTier0,
```
The caller is only interested in `CORJIT_FLAG_TIER0` flag, so we may just pass that one single flag as a bool out (better encapsulation, the caller has no business in knowing all JIT flags).